src/classes/RD2_ScheduleService.cls

```diff
@@ -412,6 +417,10 @@ public without sharing class RD2_ScheduleService {
     * @return Date The Next Donation Date
     */
     private Date getNextDonationDateGreaterThan(RecurringDonationSchedule__c schedule, Date referenceDate) {
+        if (schedule.InstallmentFrequency__c == null || schedule.InstallmentFrequency__c < 1) {
```

Comment:
Instead of checking the Installment Frequency here, can we check it in the `isActive()` method?
@npsp-reedestockton , do you see any issue with checking if the valid schedule is a schedule having Installment Frequency defined?

Reply:
From reading the code, it seem like moving the check to isActive() will be a better change.

Reply:
I don't really see anything wrong with that, but shouldn't validation be happening in the trigger?

Reply:
It does, this is the case when the client org has a trigger setting the Installment Frequency to null. Since trigger order is not guaranteed, the result is not deterministic so there can be a record having Installment Frequency being null. If so, creating schedules and Installments component fail.

Reply:
Also, after adding the error to the record, the trigger will still run from start to finish and end up throwing an exception due to bad data. The validation in ScheduleServices class further prevent the null pointer exception and divided by 0 exception

Reply:
Thanks for the explanation!